src/web/lib/contributions.tsx

```diff
@@ -0,0 +1,92 @@
+import { getCookie } from '@root/src/web/browser/cookie';
+
+// User Atributes API cookies (dropped on sign-in)
+export const HIDE_SUPPORT_MESSAGING_COOKIE = 'gu_hide_support_messaging';
+export const RECURRING_CONTRIBUTOR_COOKIE = 'gu_recurring_contributor';
+export const ONE_OFF_CONTRIBUTION_DATE_COOKIE = 'gu_one_off_contribution_date';
+
+// Support Frontend cookies (dropped when contribution is made)
+export const SUPPORT_RECURRING_CONTRIBUTOR_MONTHLY_COOKIE =
+    'gu.contributions.recurring.contrib-timestamp.Monthly';
+export const SUPPORT_RECURRING_CONTRIBUTOR_ANNUAL_COOKIE =
+    'gu.contributions.recurring.contrib-timestamp.Annual';
+export const SUPPORT_ONE_OFF_CONTRIBUTION_COOKIE =
+    'gu.contributions.contrib-timestamp';
```

Comment:
Not sure if this is a comment for this PR or not but reading cookies is Sloooow. It's also expensive sending cookies up and down the network stack. Is there benefit in considering using fewer cookies and/or using local storage?

Reply:
Yeah, unfortunately we are constrained to the interface that Frontend uses (as we need to share the state across platforms). Once migrated we can definitely play around with a better approach.

Reply:
I had a quick chat with Tom about this over lunch. How about we create our own object and store it in LS and then, when reading:

We first look in localStorage to see if it exists, if so: continue (happy path)
If it isn't there, we then move to check cookies

Then, when writing:

We write to LS _and_ cookies so that we maintain support for the legacy frontend approach.

This allows us to move forward with a more performant solution whilst still supporting the legacy code. I'd suggest abstracting this logic into a module

Reply:
We actually don't want DCR having to own this logic to read and interpret Contribution cookies. We dream of a world where Slot Machin reduces the amount of code on the platform, instead of adding to it. Soon enough this will be owned by Slot Machine/Contributions in some way (perhaps part of the client lib?) so not keen on adding that LS/Cookie layer just yet, particularly when it involves fairly involved Frontend work. I think it's a good suggestion, possibly worth doing if we can prove it's faster to read from LS than cookies, and that we can manage the risk of doing this work on Frontend and everything that comes with it.

Reply:
@oliverlloyd just to assure you your suggestion isn't being ignored ;) I added a card to our board to consider this work more carefully when we move this logic to the client library.